sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs

```diff
@@ -912,16 +931,16 @@ private void ThrowIfSessionLockLost()
             {
                 amqpRequestMessage.AmqpMessage.ApplicationProperties.Map[ManagementConstants.Request.AssociatedLinkName] = receiveLink.Name;
             }
-            amqpRequestMessage.Map[ManagementConstants.Properties.LockTokens] = new[] { new Guid(lockToken) };
+            amqpRequestMessage.Map[ManagementConstants.Properties.LockTokens] = new List<Guid>(1) { new Guid(lockToken) };
 
             AmqpResponseMessage amqpResponseMessage = await ExecuteRequest(
                 timeout,
                 amqpRequestMessage).ConfigureAwait(false);
 
             if (amqpResponseMessage.StatusCode == AmqpResponseStatusCode.OK)
             {
-                IEnumerable<DateTime> lockedUntilUtcTimes = amqpResponseMessage.GetValue<IEnumerable<DateTime>>(ManagementConstants.Properties.Expirations);
-                lockedUntil = lockedUntilUtcTimes.First();
+                DateTime[] lockedUntilUtcTimes = amqpResponseMessage.GetValue<DateTime[]>(ManagementConstants.Properties.Expirations);
```

Comment:
If nothing is returned then now an index out of bound would be raised instead of an InvalidOperation exception. Happy to roll back if you don't like this but technically this is the last LINQ usage that is not necessary under the assumption GetValue allows me to get the date times as an array (again I tried to do the necessary research in the AMQP lib but couldn't find anything)